docs/datamodel/mutation_rewrites.rst

```diff
@@ -0,0 +1,266 @@
+        modified: <datetime>'2023-04-05T13:25:04.119641Z',
+      },
+    }
+
+.. note::
+
+    Each property may have a single ``insert`` and a single ``update`` mutation
+    rewrite rule, or they may have a single rule that covers both.
+
+Special values
+==============
+
+Inside the rewrite rule's expression, you have access to a few special values:
+
+* ``__subject__`` refers to the new value being set for the property
+* ``__specified__`` is a named tuple with a key for each pointer in the type
+  and a boolean value indicating whether this value was explicitly set in the
+  mutation
+* ``__old__`` refers to the previous value in update-only mutation rewrite
+  expressions
+
+Here are some examples of the special values in use. Maybe your blog hosts
+articles about particularly controversial topics. You could use ``__subject__``
+to enforce a "cooling off" period before publishing a blog post:
+
+.. code-block:: sdl
+
+    type Post {
+      ...
+      publish_time: datetime {
+        rewrite insert, update using (
+          __subject__.publish_time ?? datetime_of_statement() +
+          cal::to_relative_duration(days := 10)
+        )
+      }
+      ...
+    }
+
+Here we take the post's ``publish_time`` if set or the time the statement is
+executed and add 10 days to it. That should give our authors time to consider
+if they want to make any changes before a post goes live.
+
+You can omit ``__subject__`` in many cases and achieve the same thing:
+
+.. code-block:: sdl-diff
+
+      type Post {
+        ...
+        publish_time: datetime {
+          rewrite insert, update using (
+    -       __subject__.publish_time ?? datetime_of_statement() +
+    +       .publish_time ?? datetime_of_statement() +
+            cal::to_relative_duration(days := 10)
+          )
+        }
+        ...
+      }
+
+but only if the path prefix has not changed. In the following schema, for
+example, the ``__subject__`` in the rewrite rule is required since, in the
+context of the nested ``select`` query, the leading dot resolves from the
+``User`` path:
+
+.. code-block:: sdl
+
+    type Post {
+      ...
+      author_email: str;
+      author_name: str {
+        rewrite insert, update using (
+          (select User {name} filter .email = __subject__.author_email).name
+        )
+      }
+    }
+    type User {
+      name: str;
+      email: str;
+    }
+
+.. note::
+
+    Learn more about how this works in our documentation on :ref:`path
+    resolution <ref_eql_path_resolution>`.
+
+Using ``__specified__``, we can override ``modified`` with a specified value if
+the user provided one in the query while still using our default value
+(``datetime_of_statement()``) if they didn't:
+
+.. code-block:: sdl
+
+    type Post {
+      ...
+      modified: datetime {
+        rewrite update using (
+          datetime_of_statement()
+          if not __specified__.mtime
+          else .mtime
+        )
+      }
+      ...
+    }
```

Comment:
A more basic example would be updating `modified` only if `title` was changed:

```
      modified: datetime {
        rewrite update using (
          datetime_of_statement()
          if __specified__.title
          else __old__.modified 
        )
      }
```

Just a suggestion, feel free to dismiss.